Move larger blocks of Api::to_tokens into util functions

DevinR528 opened the pull request:
Add trybuild tests and start work on #4 

If you want I can break this up into 2 PR's I was planning on doing that but finding good PR boundaries wasn't obvious.

jimmycuadra commented:
Just needs conflicts resolved, then I'll merge. Thanks!

jimmycuadra commented:
Looks like another conflict needs to be resolved from another PR I merged first. Sorry!

DevinR528 committed:
Fix typo in response.rs docs

DevinR528 committed:
Refactor Api::to_tokens by moving large chunks into helper functions

Return compiler error when path has invalid chars. Add trybuild tests
for ruma_api! macro.

DevinR528 commented on ruma-api-macros/src/api/request.rs:
```diff
@@ -209,26 +212,13 @@ impl TryFrom<RawRequest> for Request {
+                                    &field,
+                                    &mut newtype_body_field,
+                                    || RequestFieldKind::NewtypeBody,
+                                    || RequestFieldKind::NewtypeRawBody,
```
Is there any way to make this function generic but a bit less funky, or is the param `fn() -> T` not terrible?

jplatte commented:
> If you want I can break this up into 2 PR's I was planning on doing that but finding good PR boundaries wasn't obvious.

What was the issue? Having tests that pass on the original code and refactoring afterwards would be nice. Also, tests are failing, seemingly related to #100 (though tests passed there...).

DevinR528 commented:
#104 makes this obsolete.
